google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java

```diff
@@ -272,13 +275,40 @@ public TableResult getQueryResults(QueryResultsOption... options)
     }
 
     TableId table = ((QueryJobConfiguration) getConfiguration()).getDestinationTable();
-    // TODO(pongad): merge options?
+
+    List<QueryResultsOption> waitOptions = new ArrayList<>();
+    waitOptions.add(QueryResultsOption.pageSize(0L));
+    List<TableDataListOption> listOptions = new ArrayList<>();
+    boolean hasWaitTime = false;
+    for (QueryResultsOption option : options) {
+      switch (option.getRpcOption()) {
+        case MAX_RESULTS:
+          listOptions.add(TableDataListOption.pageSize((Long) option.getValue()));
+          break;
+        case PAGE_TOKEN:
+          listOptions.add(TableDataListOption.pageToken((String) option.getValue()));
+          break;
+        case START_INDEX:
+          listOptions.add(TableDataListOption.startIndex((Long) option.getValue()));
+          break;
+        case TIMEOUT:
+          hasWaitTime = true;
+          waitOptions.add(QueryResultsOption.maxWaitTime((Long) option.getValue()));
+          break;
+      }
+    }
+    if (!hasWaitTime) {
+      waitOptions.add(QueryResultsOption.maxWaitTime(Duration.ofMinutes(1).toMillis()));
```

Comment:
I think the API default is 10 minutes. Is there a reason to explicitly set it if it's not provided?

Reply:
I did it to keep existing behavior, but you're right that it seems odd to second-guess the API. Removed.